File: Power Query/300-399/320/PQ 320.py
import pandas as pd

path = "300-399/320/PQ_Challenge_320.xlsx"

input = pd.read_excel(path, usecols="A:C", nrows=13)
test = pd.read_excel(path, usecols="E:I", nrows=4).rename(columns=lambda c: c.replace('.1', ''))

input['Customer'] = input['Customer'].ffill()
input = input[input['Customer'] != 'Total']

result = input.pivot(index='Customer', columns='Quarter', values='Amount')
result.columns = [f"{col} Amount" for col in result.columns]
result = result.reset_index()

print(result.equals(test)) # True

File: Power Query/300-399/320/PQ 320.R
library(tidyverse)
library(readxl)

path = "Power Query/300-399/320/PQ_Challenge_320.xlsx"
input = read_excel(path, range = "A1:C13")
test  = read_excel(path, range = "E1:I5")

result = input %>%
  fill(Customer) %>%
  filter(Customer != "Total") %>%
  pivot_wider(names_from = Quarter, 
              names_glue = "{Quarter} {.value}", 
              values_from = Amount)

all.equal(result, test) # TRUE
